addons/survey/populate/survey_survey.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.tools import populate
+
+class Survey(models.Model):
+    _inherit = 'survey.survey'
+    _populate_sizes = {'small': 10, 'medium': 50, 'large': 100}
+
+    def _populate_factories(self):
+        random = populate.Random('survey_survey')
+        highest_code = int(self._read_group([], [], ['session_code:max'])[0][0])
+        user_ids = self.env['res.users'].search([]).ids
+        user_range = [_ for _ in range(0, 11)]
+        question_range = [_ for _ in range(5, 21)]
+        answer_range = [_ for _ in range(1, 5)]
+
+        def get_session_code(values=None, counter=None, **kwargs):
+            """Assign a unique number that is greater than the largest session
+            code already stored in DB"""
+            return highest_code + counter + 1
+
+        def get_question_and_page_ids(values=None, **kwargs):
+            """Create random number of questions and question's answers"""
+            return [
+                fields.Command.create({
+                    'title': f"Question {i + 1} for {values['title']}",
+                    'question_type': 'multiple_choice',
+                    'suggested_answer_ids': [fields.Command.create({
+                        'value': f"Answer {j + 1} of question {i + 1} for {values['title']} "
+                    }) for j in range(random.choice(answer_range))],
+                })
+                for i in range(random.choice(question_range))
```

Comment:
* You can start `range` at 1 to avoid `+1`s
* `i` and `j` and not meaningful names; it's better to be explicit.
```suggestion
                fields.Command.create({
                    'title': f"Question {question_idx} for {values['title']}",
                    'question_type': 'multiple_choice',
                    'suggested_answer_ids': [
                        fields.Command.create({
                            'value': f"Answer {answer_idx} of question {question_idx} for {values['title']}"
                        }) for answer_idx in range(1, random.choice(answer_range)+ 1)
                    ],
                })
                for question_idx in range(1, random.choice(question_range) + 1)
```